lib/resty/core/uri.lua

```diff
@@ -53,22 +55,41 @@ elseif subsystem == "stream" then
 end
 
 
-ngx.escape_uri = function (s)
+ngx.escape_uri = function (s, esc_type)
     if type(s) ~= 'string' then
         if not s then
             s = ''
+
         else
             s = tostring(s)
         end
     end
+
+    if esc_type == nil then
+        esc_type = NGX_ESCAPE_URI_COMPONENT
+
+    else
+        if type(esc_type) ~= 'number' then
+            error("\"type\" is not number", 3)
+        end
+
+        if esc_type < NGX_ESCAPE_URI or esc_type > NGX_ESCAPE_MAIL_AUTH then
```

Comment:
If we already check the argument on the lua-resty-core level then we don't need to check it again on the C level in lua-nginx-module.